### PR TITLE
Fixed Gameplay::update()

### PR DESCRIPTION
**Pull request by AHUMANTHATISHUMAN**
`erase()` invalidates the iterator being erased, so that it can no longer be incremented. This results in an error in debug mode, and possibly other behaviours in release mode.

### CODE REVIEW

**Comment by Hopson97** [comment]
Ah, this loop was adapted from a regular for loop, brain fart meant it wasn't correct. 

However, this is not entirely correct. While you are right about the iterator not being incremented in the loop itself, passing the iterator to the erase function will infact increment the iterator.

Otherwise, an if statement that does the increment will suffice.

File: src/client/gameplay.cpp [hunk 1]
```diff
@@ -183,13 +183,14 @@ void Gameplay::update()
...
+    for (auto itr = chunks.begin(); itr != chunks.end();) {
         const auto &[position, chunk] = *itr;
-        if (m_chunkRenders.find(position) == m_chunkRenders.end()) {
-            if (chunkMgr.hasNeighbours(position)) {
-                m_chunkRenders.emplace(position, makeChunkMesh(*chunk));
-                chunks.erase(itr);
-            }
+        if (m_chunkRenders.find(position) == m_chunkRenders.end() && chunkMgr.hasNeighbours(position)) {
+            m_chunkRenders.emplace(position, makeChunkMesh(*chunk));
+            chunks.erase(itr++);
```
Comment (Hopson97) [inline src/client/gameplay.cpp]: https://en.cppreference.com/w/cpp/container/vector/erase

```suggestion
            chunks.erase(itr);
```

Comment (AHUMANTHATISHUMAN) [reply src/client/gameplay.cpp]: Isn't `chunks` and `std::unordered_map`?
```C++
struct ClientState {
    std::array<Entity, 512> entities;
    ChunkPositionMap<Chunk *> chunks;
    ChunkManager chunkManager;

    EngineStatus status = EngineStatus::Ok;
};
```
https://en.cppreference.com/w/cpp/container/unordered_map/erase

Comment (Hopson97) [reply src/client/gameplay.cpp]: Yes it is, my mistake :)

But the same rules applies to unordered_map

https://en.cppreference.com/w/cpp/container/unordered_map/erase

Comment (AHUMANTHATISHUMAN) [reply src/client/gameplay.cpp]: No, it shouldn't it be `itr = chunks.erase(itr);`, like in the example?

Comment (Hopson97) [reply src/client/gameplay.cpp]: Yep my mistake again.

I shouldn't use github when I woke up 10 minutes ago haha

Comment (Hopson97) [inline src/client/gameplay.cpp]:
```suggestion
            itr = chunks.erase(itr);
```

**Pull request merged**